c10/core/impl/alloc_cpu.cpp

```diff
@@ -81,6 +88,15 @@ void* alloc_cpu(size_t nbytes) {
       " (",
       strerror(err),
       ")");
+  // enable thp (transparent huge pages) for larger buffers
+  if (nbytes >= gAlloc_threshold_thp) {
+      int ret = madvise(data, nbytes, MADV_HUGEPAGE);
+      if (ret != 0) {
+#ifdef DEBUG
+          fprintf(stderr, "thp madvise failed with %s\n", strerror(errno));
```

Comment:
Hmm, is spewing to stderr in debug mode really the right way to do this :/ Maybe we should do a `TORCH_WARN_ONCE` here

Reply:
thanks @ezyang , I've added TORCH_WARN_ONCE and also addressed the linter warning.